Backend restructure

josh-ja-walker opened the pull request.

aniket1101 commented on backend/src/commands.rs:
```diff
@@ -0,0 +1,156 @@
+
+#[derive(Debug, serde::Serialize)]
+enum SaveTreeError {
+    LockFailed,
```
Change to `LockingError` for consistency with line 155

aniket1101 commented on backend/src/commands.rs:
```diff
@@ -0,0 +1,156 @@
+
+#[derive(Debug, serde::Serialize)]
+enum FetchChildrenError {
+    LockFailed,
```
Note that `LockingError` is used elsewhere.

josh-ja-walker replied:
Updated all errors to use useful names - Error::LockFailed, Error::SerialiseFailed, etc...

aniket1101 commented on backend/src/state.rs:
```diff
@@ -1,11 +1,51 @@
+        self.tree.as_ref()
+    }
+
+    fn setup_map(&mut self, debug_node: &DebugNode) {
```
Comment

aniket1101 commented on backend/src/state.rs:
```diff
@@ -1,11 +1,51 @@
+        debug_node.children.iter().for_each(|child| self.setup_map(child));
+    }
+
+    pub fn get_debug_node(&self, node_id: u32) -> Option<&DebugNode> {
```
Comment

aniket1101 reviewed (changes requested):
Naming and style

josh-ja-walker commented:
Moved state to separate files
Moved commands to separate files
Changed definition of AppState and StateManager
Changed StateManager to handle errors with results

josh-ja-walker commented:
Moved `ParsleyTree` to `trees/`
Renamed `StateHandle` to `ServerState`
Moved `ServerState` to `server/`

PriyanshC commented on backend/src/commands.rs:
```diff
@@ -0,0 +1,15 @@
+use tauri;
```
Unused import

PriyanshC commented on backend/src/state/app_state.rs:
```diff
@@ -0,0 +1,90 @@
+
+
+/* Unsynchronised global AppState */
+struct AsyncAppState {
```
Async is a misleading name

PriyanshC replied:
My ideas:
`ProtectedAppState` for the Mutex wrapper on an `AppState`
`AppState` wrapping `AppStateData`
`AppState` wrapping `AppStateInternal` (or flip this with 'external')

josh-ja-walker replied:
I'll take option number 3 pls

PriyanshC commented on backend/src/trees/saved_tree.rs:
```diff
@@ -66,16 +66,32 @@ impl SavedTree {
+    pub const RAW_TREE: &str = ""; //TODO
+
+    pub fn test_tree() -> SavedTree {
+        todo!()
```
Informative message in the `todo!()` macro

PriyanshC commented on backend/src/commands/save.rs:
```diff
@@ -0,0 +1,114 @@
+}
+
+#[derive(Debug, serde::Serialize)]
+pub enum SaveTreeError {
```
Clippy warns that these all have the same suffix. Rename some or `#[allow]` it

PriyanshC commented on backend/src/trees/parsley_tree.rs:
```diff
@@ -1,4 +1,4 @@
-use crate::{DebugNode, DebugTree};
+use crate::trees::{DebugNode, DebugTree};
```
Be consistent with the other tree files, use `super::`

PriyanshC commented on backend/src/trees/debug_tree.rs:
```diff
@@ -85,3 +85,44 @@ impl DebugNode {
+                true,
+                Some(0),
+                String::from("Test"),
+                vec![],
```
`Vec::new()` is better than the empty macro
